src/mokernels/slfm.jl

```diff
@@ -0,0 +1,67 @@
+@doc raw"""
+    LatentFactorMOKernel(
+        g::AbstractVector{<:Kernel}, 
+        e::AbstractVector{<:Kernel}, 
+        A::AbstractMatrix
+        )
+
+A semiparametric kernel for problems involving multiple response variables.
+
+``k((x, p), (y, p)) = k_p(x, y) = \Sum^{Q}_{q=1} A_{pq}g_q(x, y) + e_p(x, y)``
+
+# Arguments
+- `g::AbstractVector{<:Kernel}`: an array of kernels
+- `e::AbstractVector{<:Kernel}`: an array of kernels
+- `A::AbstractMatrix`: an matrix of weights for the kernels of shape (length(e), length(g))
+
+
+# Reference:
+- [Seeger, Teh, and Jordan (2005)](https://infoscience.epfl.ch/record/161465/files/slfm-long.pdf)
+
+"""
+struct LatentFactorMOKernel{
+    Tg <:AbstractVector{<:Kernel}, 
+    Te <:AbstractVector{<:Kernel}, 
+    TA <: AbstractMatrix
+    } <: Kernel
+    g::Tg
+    e::Te
+    A::TA
+    function LatentFactorMOKernel(
+        g::AbstractVector{<:Kernel}, 
+        e::AbstractVector{<:Kernel}, 
+        A::AbstractMatrix
+        )
+        @assert (length(e), length(g)) == 
+            size(A) "Size of A not compatible to the given array of kernels"
+        return new{typeof(g), typeof(e), typeof(A)}(g, e, A)
+    end
+end
+
+function (κ::LatentFactorMOKernel)(x::Tuple{Vector, Int}, y::Tuple{Vector, Int})
```

Comment:
Maybe it would be more convenient and more readable to write
```suggestion
function (κ::LatentFactorMOKernel)((x, px)::Tuple{Vector, Int}, (y, py)::Tuple{Vector, Int})
```
and then use `x` and `px` instead of `first(x)` and `last(x)` in the function definition (and the same goes for `y` and `py`, of course).